### DISCUSSION TITLE
[Question]: Using the new review items for notifications

### DISCUSSION THREAD

**Discussion by ofcedwards** [blakeblackshear/frigate#11727] [Ask A Question] [answered]
### What is your question:

I've been testing out the new ````frigate/review```` mqtt topic to generate notifications in Home Assistant and have found a few things that don't quite work as expected (I'm currently running 0.14.0-bccffe6).

1.  Using ````FRIGATE_URL:5000/review?id=REVIEW_ID```` as a link provides a different experience than clicking on a review item from the Frigate UI. Review items clicked from the UI always start playback earlier than if the same review item is viewed by manually navigating to it through the url. I believe this is related to a const called ````REVIEW_PADDING```` that looks to be in play when a user clicks a review item. From what I can tell, this const backs up the start time by two seconds. Looking through the code, I don't see anything similar for when review items are viewed by url. This two second difference has been repeatable for me across every review item I've tested (both navigating by url to view the item and clicking on it in the ui). The two seconds is also long enough that it makes the difference between seeing the event actually start and seeing it in progress (or if a car is going by, seeing the car disappear as soon as playback starts). I'd love to use this feature because having all events merged into one review item is excellent in 0.14.0, but having to skip back at the start of every video has made me switch back to getting the event clip through the api.
2. Relating to the new way all detected objects are merged together. I've set Home Assistant to only notify on "alerts" from Frigate, but alerts will include any other object detected along with what I have set (people in a certain zone). I have no issue with this because it's much better to have one section of video to playback to capture all events, but I'm finding it difficult to grab a snapshot of the object that caused the alert. Object detections are now provided in mqtt as list, but pulling from the first item in the list does not consistently provide the desired thumbnail (a person, in my case). At times, I get a thumbnail of the person triggering the alert, but other times I get a thumbnail of a car that was walked in front of or a car driving by on the street. The thumbnails in the review section of the ui seem to constantly be what I'd want, but the .webp images to do not show up in notifications via Home Assistant on my iPhone (17.5.1). If there's some way to sort out the event id of what caused the detection from detection to alert, I'd be very grateful to know. Also, if anyone is getting .webp images to work on an iPhone, I'd love to know how.
3. Occasionally, notifications will be significantly delayed as the detection in Frigate is still in progress. The worst I've experienced was approximately a two minute delay between when the person was detected to when an alert was sent. It seemed that Frigate struggled to decide that it should be an alert (the person was detected when the appeared, but the clip was in the detection page, not the alert page). I believe this had to do with the detection starting with a car outside the zone, then entering the zone, then a person getting out of it (which should finally change it to an alert). I can only guess that somewhere in that interaction, Frigate got bogged down in determining the classification and only decided when it finalized the review item. I run Frigate in docker on unraid with a coral, so there was definitely not a hardware issue.

Any guidance on these three items would be greatly appreciated. I'm loving the new ui in 0.14.0 and the changes to the way events are reviewed has made checking over footage much more efficient. Frigate has gone from something I'd mostly use in Home Assistant to a standalone "app" I have saved to the home screen on my phone. I'd love to use some of these new changes in the way I receive notifications, but the combination of all three items has me sticking with the old method for now.

**Comment by NickM-27** [comment]
1. I agree, will adjust
2. I would suggest using the preview gif as the image in the notification instead of a single object's snapshot
3. I've not seen this. There is no real way for it to bog down, perhaps there was just not an update for some reason. Will see if others report the same

**Reply by ofcedwards** [reply]
1. I see the commit has already been done. Thanks for the quick fix on that!
2. I've got this switched over in Home Assistant using ````/api/review/{{id}}/preview````. I'm assuming this what you were referencing? I don't think this is mentioned in the beta docs, but it looks to generate a preview gif based on the review id. The only issue now is that gifs don't seem to play on Apple Watch and for some reason, they don't work at all when expanding a Home Assistant notification on iPhone (says "failed to load attachment / unrecognized attachment type"). The gif is also of the entire frame, which isn't optimal for smaller viewing (but this isn't the biggest deal). Do you have any other recommendations for something akin to the old event thumbnails for use on iPhone/Apple Watch (truthfully, mostly on Apple Watch for quickly determining if I need to pull up the camera or if it's a known/expected person)?
3. I agree this might be a one-off issue. The review item in question was a detection of me getting out of my car, so I should be able to easily replicate the scenario and see if it occurs again. I'll report back if it does and see if I can find a way to consistently replicate the issue.

**Reply by ofcedwards** [reply]
1. Solved!
2. Further testing has all filetypes (gif/webp) working on iPhone, but not Apple Watch. I can concede that my use case may be outside what's typical and I've found solutions that work for me (converting webp files with ````dwebp```` and just using ````/api/<camera_name>/<label>/best.jpg```` have both returned decent results).
3. I've experienced this a few more times, but not with as severe of a delay. I've had multiple more detections that do not change to alerts until the end of the event. These have all been events that start with a car driving up in the street, pulling up into the driveway, then having a person get out. My settings for alerts require that a person is detected and that they're in the zone in my front yard/driveway area. I was able to catch one of these events in progress and verified that it was under the detections page and not the alerts page until after the event ended (with the person already being inside my house). I had a secondary notification set up on the ````frigate/events```` topic that detected the person correctly as soon as they appeared (this notification also required presence in the same zone). I'm not sure if it's relevant, but the review item thumbnail that was generated for these events showed the car in the street (I'd expect the thumbnail to include the object that made the review item an alert, but I'm not sure how those thumbnails are chosen). I'm not sure if this issue is a logic issue in determining alert vs. detection, but I have verified that frigate is detecting the object instantaneously as expected, but is not classifying the event as an alert until the event ends. If there's any sort of additional logging I can turn on to try to capture this, please let me know.
4. In trying to track down what's happening with item 3, I noticed that the ````frigate/reviews```` topic never seems to have any difference between ````before```` and ````after````. In ````frigate/events```` I can walk out of my front door (directly into a zone) but ````before```` shows no zone and ````after```` shows the zone both as current and entered. In ````frigate/reviews```` both ````before```` and ````after```` show the zone in the same scenario. I'm not sure if this is a bug or if it's working as expected with the new topic, but it seems redundant if it's the expected behavior. This is not just for the zone, it applies to every field available for ````before```` and ````after````.  Historically, I'd use the difference between those fields to filter out extraneous notifications, which I'd like to still have the option to do with the new topic.

**Reply by ofcedwards** [reply]
@NickM-27 I have been able to repeat the slow change from detection to alert multiple times now under the same circumstances. I started logging all payloads delivered from ````frigate/events```` and ````frigate/reviews```` to capture this. I'll include the relevant payloads from one such event today, which registered as an alert 38 seconds after the person was detected in the required zone. What I'm seeing through mqtt is reflected in the frigate dashboard. This alert again showed the car in the street as the thumbnail where all other alerts shows a person in the required zone. If there is something more I can log that would be of use please let me know.

````frigate/reviews```` delivered two payloads:
````
2024-06-10 13:58:54
{
  "type": "new",
  "before": {
    "id": "1718042334.145122-xypaoo",
    "camera": "front_door",
    "start_time": 1718042334.145122,
    "end_time": null,
    "severity": "detection",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718042334.145122-xypaoo.webp",
    "data": {
      "detections": [
        "1718042333.75152-c06eit"
      ],
      "objects": [
        "car"
      ],
      "sub_labels": [],
      "zones": [],
      "audio": []
    }
  },
  "after": {
    "id": "1718042334.145122-xypaoo",
    "camera": "front_door",
    "start_time": 1718042334.145122,
    "end_time": null,
    "severity": "detection",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718042334.145122-xypaoo.webp",
    "data": {
      "detections": [
        "1718042333.75152-c06eit"
      ],
      "objects": [
        "car"
      ],
      "sub_labels": [],
      "zones": [],
      "audio": []
    }
  }
}
````
and
````
2024-06-10 14:00:10
{
  "type": "end",
  "before": {
    "id": "1718042334.145122-xypaoo",
    "camera": "front_door",
    "start_time": 1718042334.145122,
    "end_time": 1718042380.700948,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718042334.145122-xypaoo.webp",
    "data": {
      "detections": [
        "1718042371.290828-jeco9g",
        "1718042333.75152-c06eit"
      ],
      "objects": [
        "car",
        "person"
      ],
      "sub_labels": [],
      "zones": [
        "front_yard"
      ],
      "audio": []
    }
  },
  "after": {
    "id": "1718042334.145122-xypaoo",
    "camera": "front_door",
    "start_time": 1718042334.145122,
    "end_time": 1718042380.700948,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718042334.145122-xypaoo.webp",
    "data": {
      "detections": [
        "1718042371.290828-jeco9g",
        "1718042333.75152-c06eit"
      ],
      "objects": [
        "car",
        "person"
      ],
      "sub_labels": [],
      "zones": [
        "front_yard"
      ],
      "audio": []
    }
  }
}
````

````frigate/events```` delivered numerous payloads, but here are the two relevant ones:
````
2024-06-10 13:59:31
{
  "before": {
    "id": "1718042371.290828-jeco9g",
    "camera": "front_door",
    "frame_time": 1718042371.290828,
    "snapshot": null,
    "label": "person",
    "sub_label": null,
    "top_score": 0.0,
    "false_positive": true,
    "start_time": 1718042371.290828,
    "end_time": null,
    "score": 0.7265625,
    "box": [
      1010,
      366,
      1152,
      876
    ],
    "area": 72420,
    "ratio": 0.2784313725490196,
    "region": [
      740,
      346,
      1280,
      886
    ],
    "stationary": false,
    "motionless_count": 0,
    "position_changes": 0,
    "current_zones": [],
    "entered_zones": [],
    "has_clip": false,
    "has_snapshot": false,
    "attributes": {},
    "current_attributes": []
  },
  "after": {
    "id": "1718042371.290828-jeco9g",
    "camera": "front_door",
    "frame_time": 1718042371.690846,
    "snapshot": {
      "frame_time": 1718042371.690846,
      "box": [
        934,
        339,
        1117,
        893
      ],
      "area": 101382,
      "region": [
        600,
        280,
        1280,
        960
      ],
      "score": 0.8203125,
      "attributes": []
    },
    "label": "person",
    "sub_label": null,
    "top_score": 0.7265625,
    "false_positive": false,
    "start_time": 1718042371.290828,
    "end_time": null,
    "score": 0.8203125,
    "box": [
      934,
      339,
      1117,
      893
    ],
    "area": 101382,
    "ratio": 0.3303249097472924,
    "region": [
      600,
      280,
      1280,
      960
    ],
    "stationary": false,
    "motionless_count": 0,
    "position_changes": 1,
    "current_zones": [],
    "entered_zones": [],
    "has_clip": false,
    "has_snapshot": true,
    "attributes": {},
    "current_attributes": []
  },
  "type": "new"
}
````
and
````
2024-06-10 13:59:32
{
  "before": {
    "id": "1718042371.290828-jeco9g",
    "camera": "front_door",
    "frame_time": 1718042371.690846,
    "snapshot": {
      "frame_time": 1718042371.690846,
      "box": [
        934,
        339,
        1117,
        893
      ],
      "area": 101382,
      "region": [
        600,
        280,
        1280,
        960
      ],
      "score": 0.8203125,
      "attributes": []
    },
    "label": "person",
    "sub_label": null,
    "top_score": 0.7265625,
    "false_positive": false,
    "start_time": 1718042371.290828,
    "end_time": null,
    "score": 0.8203125,
    "box": [
      934,
      339,
      1117,
      893
    ],
    "area": 101382,
    "ratio": 0.3303249097472924,
    "region": [
      600,
      280,
      1280,
      960
    ],
    "stationary": false,
    "motionless_count": 0,
    "position_changes": 1,
    "current_zones": [],
    "entered_zones": [],
    "has_clip": false,
    "has_snapshot": true,
    "attributes": {},
    "current_attributes": []
  },
  "after": {
    "id": "1718042371.290828-jeco9g",
    "camera": "front_door",
    "frame_time": 1718042371.909686,
    "snapshot": {
      "frame_time": 1718042371.909686,
      "box": [
        804,
        319,
        1095,
        956
      ],
      "area": 185367,
      "region": [
        540,
        220,
        1280,
        960
      ],
      "score": 0.8203125,
      "attributes": []
    },
    "label": "person",
    "sub_label": null,
    "top_score": 0.7578125,
    "false_positive": false,
    "start_time": 1718042371.290828,
    "end_time": null,
    "score": 0.8203125,
    "box": [
      804,
      319,
      1095,
      956
    ],
    "area": 185367,
    "ratio": 0.45682888540031397,
    "region": [
      540,
      220,
      1280,
      960
    ],
    "stationary": false,
    "motionless_count": 1,
    "position_changes": 1,
    "current_zones": [
      "front_yard"
    ],
    "entered_zones": [
      "front_yard"
    ],
    "has_clip": true,
    "has_snapshot": true,
    "attributes": {},
    "current_attributes": []
  },
  "type": "update"
}
````

**Reply by NickM-27** [reply]
thank you for the very detailed information, I have realized an edge case where this can happen and it is fixed in https://github.com/blakeblackshear/frigate/pull/11879

**Reply by ofcedwards** [reply]
Excellent, thank you! I've changed over to the latest dev image (dev-272a21f) and will report back if I see the issue again. I figured I just had the right set of circumstances to expose the issue. Are you able to advise if the mirrored before/after sections of the ````frigate/reviews```` payload is working as intended or if it should function like the sections in ````frigate/events````?

**Reply by NickM-27** [reply]
I don't have mirrored before / after, the automation I wrote a guide for depends on them not being mirrored or else notifications would not work.

**Reply by ofcedwards** [reply]
That's how I've historically used the payloads as well. But with the new topic, I've never seen a difference in the before/after. I specifically use this for validating new events in zones, but it applies to the whole payload. Below is an example from each topic for a person event that started out of a zone and then entered a zone. Before/after show the same zone despite the ````frigate/events```` topic correctly showing it as a new zone entry.

````frigate/reviews```` payload:
````
2024-06-11 16:26:56
{
  "type": "update",
  "before": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": null,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137595.85912-9zxp73",
        "1718137612.620278-0p6rg7"
      ],
      "objects": [
        "person",
        "car"
      ],
      "sub_labels": [],
      "zones": [
        "front_yard"
      ],
      "audio": []
    }
  },
  "after": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": null,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137595.85912-9zxp73",
        "1718137612.620278-0p6rg7"
      ],
      "objects": [
        "person",
        "car"
      ],
      "sub_labels": [],
      "zones": [
        "front_yard"
      ],
      "audio": []
    }
  }
}
````

````frigate/events```` payload:
````
2024-06-11 16:26:56
{
  "before": {
    "id": "1718137612.620278-0p6rg7",
    "camera": "front_door",
    "frame_time": 1718137614.031436,
    "snapshot": {
      "frame_time": 1718137614.031436,
      "box": [
        684,
        483,
        715,
        569
      ],
      "area": 2666,
      "region": [
        535,
        364,
        855,
        684
      ],
      "score": 0.76953125,
      "attributes": []
    },
    "label": "person",
    "sub_label": null,
    "top_score": 0.7265625,
    "false_positive": false,
    "start_time": 1718137612.620278,
    "end_time": null,
    "score": 0.76953125,
    "box": [
      684,
      483,
      715,
      569
    ],
    "area": 2666,
    "ratio": 0.36046511627906974,
    "region": [
      535,
      364,
      855,
      684
    ],
    "stationary": false,
    "motionless_count": 0,
    "position_changes": 1,
    "current_zones": [],
    "entered_zones": [],
    "has_clip": false,
    "has_snapshot": true,
    "attributes": {},
    "current_attributes": []
  },
  "after": {
    "id": "1718137612.620278-0p6rg7",
    "camera": "front_door",
    "frame_time": 1718137616.419911,
    "snapshot": {
      "frame_time": 1718137616.223911,
      "box": [
        679,
        485,
        714,
        579
      ],
      "area": 3290,
      "region": [
        525,
        369,
        845,
        689
      ],
      "score": 0.71484375,
      "attributes": []
    },
    "label": "person",
    "sub_label": null,
    "top_score": 0.75390625,
    "false_positive": false,
    "start_time": 1718137612.620278,
    "end_time": null,
    "score": 0.71484375,
    "box": [
      686,
      480,
      719,
      586
    ],
    "area": 3498,
    "ratio": 0.3113207547169811,
    "region": [
      524,
      371,
      844,
      691
    ],
    "stationary": false,
    "motionless_count": 3,
    "position_changes": 1,
    "current_zones": [
      "front_yard"
    ],
    "entered_zones": [
      "front_yard"
    ],
    "has_clip": true,
    "has_snapshot": true,
    "attributes": {},
    "current_attributes": []
  },
  "type": "update"
}
````

**Reply by NickM-27** [reply]
You can't just look at a single message, many are sent.

**Reply by ofcedwards** [reply]
I provided just the message where the change should have occurred for brevity. Below are all of the ````frigate/reviews```` payloads for that event. None of them show an empty list in the before section for zones and a listed zone in the after section as expected. The first message with zones (also where it changes to alert) show the zone in both sections.

````
2024-06-11 16:26:38
{
  "type": "new",
  "before": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": null,
    "severity": "detection",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137595.85912-9zxp73"
      ],
      "objects": [
        "car"
      ],
      "sub_labels": [],
      "zones": [],
      "audio": []
    }
  },
  "after": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": null,
    "severity": "detection",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137595.85912-9zxp73"
      ],
      "objects": [
        "car"
      ],
      "sub_labels": [],
      "zones": [],
      "audio": []
    }
  }
}
2024-06-11 16:26:54
{
  "type": "update",
  "before": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": null,
    "severity": "detection",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137595.85912-9zxp73",
        "1718137612.620278-0p6rg7"
      ],
      "objects": [
        "person",
        "car"
      ],
      "sub_labels": [],
      "zones": [],
      "audio": []
    }
  },
  "after": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": null,
    "severity": "detection",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137595.85912-9zxp73",
        "1718137612.620278-0p6rg7"
      ],
      "objects": [
        "person",
        "car"
      ],
      "sub_labels": [],
      "zones": [],
      "audio": []
    }
  }
}
2024-06-11 16:26:56
{
  "type": "update",
  "before": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": null,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137595.85912-9zxp73",
        "1718137612.620278-0p6rg7"
      ],
      "objects": [
        "person",
        "car"
      ],
      "sub_labels": [],
      "zones": [
        "front_yard"
      ],
      "audio": []
    }
  },
  "after": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": null,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137595.85912-9zxp73",
        "1718137612.620278-0p6rg7"
      ],
      "objects": [
        "person",
        "car"
      ],
      "sub_labels": [],
      "zones": [
        "front_yard"
      ],
      "audio": []
    }
  }
}
2024-06-11 16:26:56
{
  "type": "update",
  "before": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": null,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137595.85912-9zxp73",
        "1718137612.620278-0p6rg7"
      ],
      "objects": [
        "person",
        "car"
      ],
      "sub_labels": [],
      "zones": [
        "front_yard"
      ],
      "audio": []
    }
  },
  "after": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": null,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137595.85912-9zxp73",
        "1718137612.620278-0p6rg7"
      ],
      "objects": [
        "person",
        "car"
      ],
      "sub_labels": [],
      "zones": [
        "front_yard"
      ],
      "audio": []
    }
  }
}
2024-06-11 16:28:03
{
  "type": "end",
  "before": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": 1718137653.034931,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137649.834217-hdkwp0",
        "1718137595.85912-9zxp73",
        "1718137612.620278-0p6rg7"
      ],
      "objects": [
        "person",
        "car"
      ],
      "sub_labels": [],
      "zones": [
        "front_yard"
      ],
      "audio": []
    }
  },
  "after": {
    "id": "1718137598.413521-34tm00",
    "camera": "front_door",
    "start_time": 1718137598.413521,
    "end_time": 1718137653.034931,
    "severity": "alert",
    "thumb_path": "/media/frigate/clips/review/thumb-front_door-1718137598.413521-34tm00.webp",
    "data": {
      "detections": [
        "1718137649.834217-hdkwp0",
        "1718137595.85912-9zxp73",
        "1718137612.620278-0p6rg7"
      ],
      "objects": [
        "person",
        "car"
      ],
      "sub_labels": [],
      "zones": [
        "front_yard"
      ],
      "audio": []
    }
  }
}
````

**Reply by NickM-27** [reply]
Ah right, that will be caused by the same issue that was fixed

**Reply by ofcedwards** [reply]
I was on dev-c9d253a at the time of that event, which I believe should have included the fix?

**Reply by ofcedwards** [reply]
The update issue appears to be fixed. I had the same circumstances today (same car, person, and positioning) and the event was changed from detection to alert as expected. However, the mqtt payloads are still mirrored. This is on dev-6978140. I'm not sure if this section of the review maintainer may be the cause:

````
        prev_data = segment.get_data(ended=False)

        if frame is not None:
            segment.update_frame(camera_config, frame, objects)

        new_data = segment.get_data(ended=False)
````
This is starting at line 199 and is under the ````update_segment```` function. It appears that ````prev_data```` and ````new_data```` are being defined as the same thing. The function called in the middle (update_frame) appears to only relate to the image that's displayed for the segment. I could be completely wrong here because I'm not super familiar with python, but to me it appears that both variables would be equal and would result in the same output.

**Reply by NickM-27** [reply]
yeah, that was mainly to account for the image path but it is correct that some parts are set before that happens, will fix

**Reply by ofcedwards** [reply, marked as answer]
Fixed! Tested this out on the new dev image and it's showing all the expected changes now (no zone to zone, detection to alert, list of event ids changes length).

Thank you for all the help through this! 0.14 has been a huge upgrade and I'm really excited to see where frigate goes in the future. You guys are really killing it.
